File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.java
class Solution {
    public int maxProfit(int[] prices) {
        //prices=[5,1,5,6,7,1,10]
        int left = 0;
        int right = 1;
        int maxProfit = 0;
        while (left < right && right <= prices.length - 1) {
            maxProfit = Math.max(maxProfit, prices[right] - prices[left]);
            if (prices[left] < prices[right]) {
                right ++;
            } else {
                left = right;
            }

            if (left == right) {
                right ++;
            }

        }

        return maxProfit;
    }
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        
        def height(root): 
            if not root: 
                return 0 
            
            return 1 + max(height(root.left), height(root.right))
            
        def dfs(root): 
            if not root: 
                return True 

            if abs(height(root.left) - height(root.right)) > 1: 
                return False
            
            return dfs(root.left) and dfs(root.right)
        
        return dfs(root)
            